src/core_init_atmosphere/mpas_parse_geoindex.F

```diff
@@ -0,0 +1,238 @@
+module mpas_parse_geoindex
+
+   use mpas_log, only : mpas_log_write
+   use mpas_derived_types, only : MPAS_LOG_ERR, MPAS_LOG_WARN
+   use mpas_pool_routines
+
+   implicit none
+
+   private
+
+   public :: mpas_parse_index
+
+   contains
+
+   !***********************************************************************
+   !
+   !  subroutine mpas_parse_geoindex
+   !
+   !> \brief   Parse a geogrid's index file and put the results into an MPAS pool
+   !> \author  Miles A. Curry 
+   !> \date    02/20/2020
+   !> \details 
+   !> Parse an index file of a static data set into an MPAS pool, allocating
+   !> each keyword=value pair into the pool with the pool member key being 
+   !> keyword, and the value being value.
+   !>
+   !> This function can parse index files with one keyword=value pair 
+   !> per line; a "#" at the start of a line, which will cause the line to be 
+   !> ignored; or an empty line containing only a newline/return character, which
+   !> will also be ignored. Spaces or tabs before, between or after the 
+   !> keyword=value tokens are > ignored.
+   !>
+   !> If a line contains anything but the above valid syntaxes, a syntax
+   !> error will be raised and the model will fail.
+   !>
+   !> Case is ignored.
+   !>
+   !> The definitions of a keyword, which can found in section 3-53
+   !> of the WRF-AWR User's Guide, will determine the corresponding type
+   !> of that keyword. A keyword that has been assigned the wrong type
+   !> will raise a type error which cause the model to fail.
+   !>
+   !> Keywords that are not handled explicitly by this function will produce
+   !> a warning that the keyword is unrecognized.
+   !
+   !-----------------------------------------------------------------------
+   subroutine mpas_parse_index(path, geo_pool)
+
+      implicit none
+      ! Input Variables
+      character (len=StrKIND), intent(in) :: path
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
+
+      ! Local Variables
+      character (len=StrKIND) :: index_path, line, lhs, rhs
+      character (len=StrKIND) :: read_err_msg, open_msg
+      integer :: open_stat, read_stat, line_read_stat
+      integer :: i, k
+      logical :: res
+
+      character (len=StrKIND), pointer :: char_t
+      integer :: iceiling, ifloor
+      integer, pointer :: int_t
+      real, pointer :: real_t
```

Comment:
Currently, if MPAS is built with double-precision reals, we define RKIND=8 and add compiler flags to promote default reals to 64-bits; but, in future, we may remove the compiler flags. So it would be best if `real` variables were always assigned an explicit kind, e.g., `real(kind=RKIND)`.